front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/ManchetteWithSpaceTimeChart.tsx

```diff
@@ -100,16 +106,41 @@ const ManchetteWithSpaceTimeChartWrapper = ({
 
   const [waypointsPanelIsOpen, setWaypointsPanelIsOpen] = useState(false);
 
-  const [tmpSelectedTrain, setTmpSelectedTrain] = useState(selectedTrainScheduleId);
-
-  useEffect(() => {
-    setTmpSelectedTrain(selectedTrainScheduleId);
-  }, [selectedTrainScheduleId]);
+  const projectedTrainsAndOccurrences = useMemo(
+    () =>
+      projectPathTrainResult.flatMap((train) => {
+        if (isOccurrence(train.id) && train.paced) {
+          const pacedTrainId = getPacedTrainIdFromOccurrenceId(train.id);
+          const stepInMs = Duration.parse(train.paced.step).ms;
+          const occurrencesCount = Math.ceil(Duration.parse(train.paced.duration).ms / stepInMs);
```

Comment:
You can use the new function getOccurrencesNb (after your rebase)
```suggestion
          const occurrencesCount = getOccurrencesNb(train.paced);
```

Reply:
done